orangecontrib/text/vectorization/document_embedder.py

```diff
@@ -0,0 +1,170 @@
+LANGUAGES = list(LANGS_TO_ISO.values())
+
+class PretrainedEmbedder:
+    """This class is used for obtaining dense embeddings of documents in
+    corpus using fastText pretrained models from:
+    E. Grave, P. Bojanowski, P. Gupta, A. Joulin, T. Mikolov,
+    Learning Word Vectors for 157 Languages.
+    Proceedings of the International Conference on Language Resources and Evaluation, 2018.
+
+    Embedding is performed on server so the internet connection is a
+    prerequisite for using the class. Currently supported languages are:
+        - English (en)
+        - Slovenian (sl)
+        - German (de)
+
+    Attributes
+    ----------
+    language : str
+        ISO 639-1 (two-letter) code of desired language.
+    aggregator : str
+        Aggregator which creates document embedding (single
+        vector) from word embeddings (multiple vectors).
+        Allowed values are mean, sum, max, min.
+    """
+
+    def __init__(self, language: str = 'en',
+                 aggregator: str = 'mean') -> None:
+        lang_error = '{} is not a valid language. Allowed values: {}'
+        agg_error = '{} is not a valid aggregator. Allowed values: {}'
+        if language.lower() not in LANGUAGES:
+            raise ValueError(lang_error.format(language, ', '.join(LANGUAGES)))
+        self.language = language.lower()
+        if aggregator.lower() not in AGGREGATORS:
+            raise ValueError(agg_error.format(aggregator, ', '.join(AGGREGATORS)))
+        self.aggregator = aggregator.lower()
+
+        self._dim = 300
+        self._embedder = _ServerEmbedder(self.aggregator,
+                                         model_name=self.language,
+                                         max_parallel_requests=1,
+                                         server_url='',
+                                         # TODO set proper url
+                                         embedder_type='text')
+
+    def __call__(self, corpus: Corpus, copy: bool = True,
+                 processed_callback=None) -> Corpus:
+        """Adds matrix of document embeddings to a corpus.
+
+        Parameters
+        ----------
+        corpus : Corpus
+            Corpus on which transform is performed.
+        copy : bool
+            If set to True, a copy of corpus is made.
+
+        Returns
+        -------
+        Corpus
+            Corpus (original or a copy) with new features added.
+
+        Raises
+        ------
+        ValueError
+            If corpus is not instance of Corpus.
+        RuntimeError
+            If document in corpus is larger than
+            50 KB after compression.
+        """
+        if not isinstance(corpus, Corpus):
+            raise ValueError("Input should be instance of Corpus.")
+        corpus = corpus.copy() if copy else corpus
+        embs = self._embedder.embedd_data(corpus.tokens,
+                                          processed_callback=processed_callback)
```

Comment:
in this case, it is preferable to break earlier and have both parameters in new line:
```
embs= self._embedder.embedd_data(
    corpus.tokens, processed_callback=processed_callback)
```